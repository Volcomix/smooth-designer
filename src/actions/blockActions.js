//@flow
import {
  ADD_BLOCK,
  UPDATE_BLOCK_NAME,
  UPDATE_BLOCK_SIZE,
} from '../constants/actionTypes'

type AddBlockAction = { type: typeof ADD_BLOCK }

type UpdateBlockNameAction = {
  type: typeof UPDATE_BLOCK_NAME,
  id: string,
  name: string,
}

type UpdateBlockSizeAction = {
  type: typeof UPDATE_BLOCK_SIZE,
  id: string,
  width: number,
  height: number,
}

export type BlockAction =
  | AddBlockAction
  | UpdateBlockNameAction
  | UpdateBlockSizeAction

export const addBlock = (): AddBlockAction => ({ type: ADD_BLOCK })

export const updateBlockName = (
  id: string,
  name: string,
): UpdateBlockNameAction => ({ type: UPDATE_BLOCK_NAME, id, name })

export const updateBlockSize = (
  id: string,
  width: number,
  height: number,
): UpdateBlockSizeAction => ({ type: UPDATE_BLOCK_SIZE, id, width, height })
