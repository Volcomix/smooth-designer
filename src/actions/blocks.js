//@flow
import {
  ADD_BLOCK,
  DELETE_BLOCK,
  UPDATE_BLOCK_NAME,
  UPDATE_BLOCK_SIZE,
} from '../constants/actionTypes'

type AddBlockAction = { type: typeof ADD_BLOCK }

type DeleteBlockAction = { type: typeof DELETE_BLOCK, id: string }

type UpdateBlockNameAction = {
  type: typeof UPDATE_BLOCK_NAME,
  id: string,
  name: string,
}

type UpdateBlockSizeAction = {
  type: typeof UPDATE_BLOCK_SIZE,
  id: string,
  width: number,
  height: number,
}

export type BlocksAction =
  | AddBlockAction
  | DeleteBlockAction
  | UpdateBlockNameAction
  | UpdateBlockSizeAction

export const addBlock = (): AddBlockAction => ({ type: ADD_BLOCK })

export const deleteBlock = (id: string): DeleteBlockAction => ({
  type: DELETE_BLOCK,
  id,
})

export const updateBlockName = (
  id: string,
  name: string,
): UpdateBlockNameAction => ({ type: UPDATE_BLOCK_NAME, id, name })

export const updateBlockSize = (
  id: string,
  width: number,
  height: number,
): UpdateBlockSizeAction => ({ type: UPDATE_BLOCK_SIZE, id, width, height })
