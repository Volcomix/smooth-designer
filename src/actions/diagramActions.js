//@flow
import { START_LINKING, END_LINKING } from '../constants/actionTypes'

type StartLinkingAction = {
  type: typeof START_LINKING,
  fromBlock: string,
}

type EndLinkingAction = {
  type: typeof END_LINKING,
  toBlock: string,
}

export type DiagramAction = StartLinkingAction | EndLinkingAction

export const startLinking = (fromBlock: string): StartLinkingAction => ({
  type: START_LINKING,
  fromBlock,
})

export const endLinking = (toBlock: string): EndLinkingAction => ({
  type: END_LINKING,
  toBlock,
})
