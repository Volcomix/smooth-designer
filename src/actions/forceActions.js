//@flow
import { UPDATE_FORCE } from '../constants/actionTypes'
import type { Block } from '../types'

type UpdateForceAction = { type: typeof UPDATE_FORCE, blocks: Block[] }

export type ForceAction = UpdateForceAction

export const updateForce = (blocks: Block[]): UpdateForceAction => ({
  type: UPDATE_FORCE,
  blocks,
})
