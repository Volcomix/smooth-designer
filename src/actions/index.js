//@flow
import type { BlockAction } from './blockActions'
import type { ForceAction } from './forceActions'
import type { DiagramAction } from './diagramActions'

export type Action = BlockAction | ForceAction | DiagramAction
