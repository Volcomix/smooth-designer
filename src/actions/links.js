//@flow
import {
  START_LINKING,
  UPDATE_LINKING,
  END_LINKING,
  CANCEL_LINKING,
} from '../constants/actionTypes'

type StartLinkingAction = {
  type: typeof START_LINKING,
  fromId: string,
  toMouse: { x: number, y: number },
}

type UpdateLinkingAction = {
  type: typeof UPDATE_LINKING,
  toMouse: { x: number, y: number },
}

type EndLinkingAction = {
  type: typeof END_LINKING,
  toId: string,
}

type CancelLinkingAction = {
  type: typeof CANCEL_LINKING,
}

export type LinksAction =
  | StartLinkingAction
  | UpdateLinkingAction
  | EndLinkingAction
  | CancelLinkingAction

export const startLinking = (
  fromId: string,
  toMouseX: number,
  toMouseY: number,
): StartLinkingAction => ({
  type: START_LINKING,
  fromId,
  toMouse: { x: toMouseX, y: toMouseY },
})

export const updateLinking = (
  toMouseX: number,
  toMouseY: number,
): UpdateLinkingAction => ({
  type: UPDATE_LINKING,
  toMouse: { x: toMouseX, y: toMouseY },
})

export const endLinking = (toId: string): EndLinkingAction => ({
  type: END_LINKING,
  toId,
})

export const cancelLinking = (): CancelLinkingAction => ({
  type: CANCEL_LINKING,
})
