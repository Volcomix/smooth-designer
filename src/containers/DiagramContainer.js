//@flow
import type { Dispatch } from 'redux'
import { connect } from 'react-redux'
import type { State } from '../reducers'
import { getBlocks } from '../reducers/blocks'
import type { Action } from '../actions'
import { addBlock, updateBlockSize } from '../actions/blockActions'
import Diagram from '../components/Diagram'

const mapStateToProps = (state: State) => ({
  blocks: getBlocks(state.blocks),
})

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  onAddClick: () => dispatch(addBlock()),
  onSizeChange: (id: string, width: number, height: number) =>
    dispatch(updateBlockSize(id, width, height)),
})

const DiagramContainer = connect(mapStateToProps, mapDispatchToProps)(Diagram)

export default DiagramContainer
