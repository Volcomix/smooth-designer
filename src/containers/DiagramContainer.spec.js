//@flow
import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import DiagramContainer from './DiagramContainer'
import type { State } from '../reducers'
import { addBlock } from '../actions/blockActions'

const mockStore = configureStore()

const setup = () => {
  const state: State = { blocks: [{ name: 'Block', x: 0, y: 0 }] }
  const store = mockStore(state)
  const wrapper = shallow(<DiagramContainer store={store} />)
  return { wrapper, store }
}

it('renders without crashing', () => {
  const { wrapper } = setup()
  expect(wrapper).toMatchSnapshot()
})

it('adds a block when the Add button is clicked', () => {
  const { wrapper, store } = setup()
  wrapper.find('Diagram').simulate('addClick')
  expect(store.getActions()).toEqual([addBlock()])
})
