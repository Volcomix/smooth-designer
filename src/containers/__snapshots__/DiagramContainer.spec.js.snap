// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Diagram
  blocks={
    Array [
      Object {
        "name": "Block",
      },
    ]
  }
  onAddClick={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
